src/utils/utils.ts

```diff
@@ -255,6 +255,14 @@ export module _Util {
     }
 
     /**
+     * Returns true if the argument is a number, which is not NaN
+     * Numbers represented as strings do not pass this function
+     */
+    export function isValidNumber(n: any) {
```

Comment:
I was going to suggest naming this `isFiniteNumber`, but we might want to support infinite values alter, so I think `isValidNumber` is a good name.